### PR TITLE
Copy contained levels when copying levels by suffix

### PR DESCRIPTION
**Pull request by davidsbailey**
### Background

See https://github.com/code-dot-org/code-dot-org/pull/21101 which copies levels by suffix, and also copies their project template levels.

### Description

Creates copies of any contained levels when copying a level by suffix. This gets us another step closer to script copying.

### CODE REVIEW

**Comment by davidsbailey** [comment]
@Hamms , could you please review?  FYI @balderdash .

File: dashboard/app/models/levels/level.rb [hunk 1]
```diff
@@ -448,6 +448,7 @@ def concept_level?
   # Returns an array of all the contained levels
   # (based on the contained_level_names property)
   def contained_levels
+    return [] unless is_a? Blockly
```
Comment (Hamms) [inline dashboard/app/models/levels/level.rb]: this seems redundant with checking for `"contained_level_names"` below; am I misunderstanding something?

Comment (davidsbailey) [reply dashboard/app/models/levels/level.rb]: Good question, it took me some time to figure out the answer to this! :-) The check below raises when you try to look at `properties["contained_level_names"]` for non-Blockly levels, which don't have this property in their serialized attributes. I'm open to guarding against this a different way if you have one to recommend.

Comment (Hamms) [reply dashboard/app/models/levels/level.rb]: Ah, in that case I'd rather update the check below to accomodate that case. Given that `contained_level_names` should be defined as a serializedproperty, we also probably don't need to reference the properties hash directly. Maybe something like

```ruby
def contained_levels
     names = try(:contained_level_names)
     return [] unless names.present?
     names.map do |contained_level_name|
```

Comment (davidsbailey) [reply dashboard/app/models/levels/level.rb]: Thanks Elijah! That looks better.

**Pull request merged**